Api/ProfileApi.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Api;
+
+use App\Api\Dto\Profile\GetCompanionProfilePreviewListParams;
+use App\Api\Structure\PageResult;
+use App\Entity\Profile\Profile;
+
+final class ProfileApi
+{
+    public function getCompanionProfilePreviewList(
+        Profile $currentProfile,
+        GetCompanionProfilePreviewListParams $params,
+    ): PageResult {
+        $showAssignedToMyProfile = $currentProfile->getId() === $params->getAssignedToProfile()?->getId();
+        $approved = $showAssignedToMyProfile ? null : true;
+        $totalCount = $this->profileRepository->findCompanionCount($currentProfile, $params, $approved);
+
+        $viewPort = $params->getViewPort();
+        if ($totalCount <= 0) {
+            if ($viewPort === null) {
+                return new PageResult([], 0, $params->getLimit());
+            }
+            $viewPort->increase();
```

Comment:
To se mě vůbec nelíbí. Programátorovi ani nemůže dojít, že tenhle řádek k něčemu je

Vytáhl jsem si viewport z $params, tím increase to nějak modifikuju a pak předávám params s tím modifikovaným viewPortem.